### PR TITLE
IDEMPIERE-5925: Method Dialog.info with callback

### PR DESCRIPTION
**Pull request by nmicoud**
https://idempiere.atlassian.net/browse/IDEMPIERE-5925

# Pull Request Checklist

- [X] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [X] My code follows the best practices of this project
- [X] I have performed a self-review of my own code
- [X] My code is easy to understand and review. 
- [X] I have checked my code and corrected any misspellings
- [ ] In hard-to-understand areas, I have commented my code.
- [X] My changes generate no new warnings
### Tests
- [X] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java [hunk 1]
```diff
@@ -591,17 +591,31 @@ public static void info(int windowNo, String adMessage, String additionalMessage
...
+    /**
+     *  Display dialog with information icon.
+     *
+     *  @param  windowNo    Number of Window
+     *  @param  adMessage   Message to be translated
+     *  @param  additionalMessage     Additional message
+     *  @param	title		If none then one will be generated
+     *  @param  callback
+     */
+
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java:606]: please remove the empty line between javadoc and method.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java [hunk 2]
```diff
@@ -591,17 +591,31 @@ public static void info(int windowNo, String adMessage, String additionalMessage
...
+    public static void info(int windowNo, String adMessage, String additionalMessage, String title, Callback<Integer> callback) {
+    	if (logger.isLoggable(Level.INFO)) logger.info(adMessage + " - " + additionalMessage);
+
+    	if (CLogMgt.isLevelFinest()) {
+    		Trace.printStack();
+    	}
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java:612]: this is harmless but I don't see when do we need this at all, I would prefer to just remove it.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java [hunk 3]
```diff
@@ -591,17 +591,31 @@ public static void info(int windowNo, String adMessage, String additionalMessage
...
-        
-        String dialogTitle = getDialogTitle(title, windowNo);
+    /**
+     *  Display dialog with information icon.
+     *
+     *  @param  windowNo    Number of Window
+     *  @param  adMessage   Message to be translated
+     *  @param  additionalMessage     Additional message
+     *  @param	title		If none then one will be generated
+     *  @param  callback
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java:604]: please note that the callback is optional

**Review by hengsin — CHANGES_REQUESTED** [review]

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java (hunk 1 above)
Comment (nmicoud) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java]: done

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java (hunk 2 above)
Comment (nmicoud) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java]: done

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java (hunk 3 above)
Comment (nmicoud) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java]: done

**Review by hengsin — APPROVED** [review]
looks good

**Pull request merged**